### ISSUE TITLE
Sentry does not play along with Paramiko's SSHException

### ISSUE THREAD

**Issue by Lilja** [getsentry/sentry-python#639]
Hello!

We're using [paramiko]() for one of our projects. We sometimes get an exception that the library is throwing: `paramiko.ssh_exception.SSHException`. 

Basically, sentry is publishing each line of the traceback as individual errors.

![image](https://user-images.githubusercontent.com/6134511/75791068-5446d200-5d6c-11ea-8de9-7ccdcd861ba1.png)

I've read [this](https://stackoverflow.com/questions/58110732/pysftp-paramiko-exceptions-leaking-into-stderr) stack overflow over and over, trying to figure out what is wrong and I've got a working example.

```python
lilja in ~/c/u/customs-readjob-setup on new-lambda-handler λ cat paramiko_test.py
#!/usr/bin/env python
import sys
import socket
import paramiko
import logging

logger = logging.getLogger(__name__)

import sentry_sdk

sentry_sdk.init(dsn="")

_sys_stderr = sys.stderr

host = ""
username = ""
password = ""
port = ""

class StdOut(object):
    def __init__(self, stream, _type):
        self.txtctrl = stream
        self.type = _type
    def write(self, string):
        self.txtctrl.write(f"{self.type} -> {string}")

    def flush(self):
        self.txtctrl.flush()

sys.stdout = StdOut(sys.stdout, "out")
sys.stderr = StdOut(sys.stderr, "err")

try:
    logger.info(f"Logging in to {host} as {username}")
    transport = paramiko.Transport((host, port))
    transport.banner_timeout = 500
    transport.auth_timeout = 500
    transport.connect(None, username=username, password=password)
    k = paramiko.SFTPClient.from_transport(transport)
except paramiko.ssh_exception.BadAuthenticationType as e:
    print(f"Bad authentication type. {e!r}")
    exit(1)
except (socket.error, socket.gaierror, paramiko.ssh_exception.SSHException) as e:
    if "Error reading SSH protocol banner" in str(e):
        print(
            "Error reading ssh protocol banner. This might be a temporary error, "
            + "or the server is a FTP(not SFTP) server."
        )
        exit(1)
    raise
```

Output: 
```
err -> Exception: Error reading SSH protocol banner
err -> Traceback (most recent call last):
err ->   File "/Users/lilja/code/umbrella/customs-readjob-setup/.venv/lib/python3.8/site-packages/paramiko/transport.py", line 2211, in _check_banner
err ->     buf = self.packetizer.readline(timeout)
err ->   File "/Users/lilja/code/umbrella/customs-readjob-setup/.venv/lib/python3.8/site-packages/paramiko/packet.py", line 380, in readline
err ->     buf += self._read_timeout(timeout)
err ->   File "/Users/lilja/code/umbrella/customs-readjob-setup/.venv/lib/python3.8/site-packages/paramiko/packet.py", line 622, in _read_timeout
err ->     raise socket.timeout()
err -> socket.timeout
err ->
err -> During handling of the above exception, another exception occurred:
err ->
err -> Traceback (most recent call last):
err ->   File "/Users/lilja/code/umbrella/customs-readjob-setup/.venv/lib/python3.8/site-packages/paramiko/transport.py", line 2039, in run
err ->     self._check_banner()
err ->   File "/Users/lilja/code/umbrella/customs-readjob-setup/.venv/lib/python3.8/site-packages/paramiko/transport.py", line 2215, in _check_banner
err ->     raise SSHException(
err -> paramiko.ssh_exception.SSHException: Error reading SSH protocol banner
err ->
out -> Error reading ssh protocol banner. This might be a temporary error, or the server is a FTP(not SFTP) server.out ->
err -> Sentry is attempting to send 18 pending error messages
err -> Waiting up to 2 seconds
err -> Press Ctrl-C to quit
```
Notice how sentry wants to send **18** error messages! I think that might be a bit much.

According to the stack overflow post, the reason is because `paramiko.Transport` is doing its work in a thread, which is posting its errors to stderr. The proposed solution is to pipe errors to stdout.

```python
try:
    logger.info(f"Logging in to {host} as {username}")
    sys.stderr = sys.stdout
    transport = paramiko.Transport((host, port))
    transport.banner_timeout = 500
    transport.auth_timeout = 500
    transport.connect(None, username=username, password=password)
    k = paramiko.SFTPClient.from_transport(transport)
except paramiko.ssh_exception.BadAuthenticationType as e:
    print(f"Bad authentication type. {e!r}")
    exit(1)
except (socket.error, socket.gaierror, paramiko.ssh_exception.SSHException) as e:
    if "Error reading SSH protocol banner" in str(e):
        print(
            "Error reading ssh protocol banner. This might be a temporary error, "
            + "or the server is a FTP(not SFTP) server."
        )
        exit(1)
    raise
finally:
    sys.stderr = _sys_stderr
```

Which brings the output:
```
out -> Exception: Error reading SSH protocol banner
out -> Traceback (most recent call last):
out ->   File "/Users/lilja/code/umbrella/customs-readjob-setup/.venv/lib/python3.8/site-packages/paramiko/transport.py", line 2211, in _check_banner
out ->     buf = self.packetizer.readline(timeout)
out ->   File "/Users/lilja/code/umbrella/customs-readjob-setup/.venv/lib/python3.8/site-packages/paramiko/packet.py", line 380, in readline
out ->     buf += self._read_timeout(timeout)
out ->   File "/Users/lilja/code/umbrella/customs-readjob-setup/.venv/lib/python3.8/site-packages/paramiko/packet.py", line 622, in _read_timeout
out ->     raise socket.timeout()
out -> socket.timeout
out ->
out -> During handling of the above exception, another exception occurred:
out ->
out -> Traceback (most recent call last):
out ->   File "/Users/lilja/code/umbrella/customs-readjob-setup/.venv/lib/python3.8/site-packages/paramiko/transport.py", line 2039, in run
out ->     self._check_banner()
out ->   File "/Users/lilja/code/umbrella/customs-readjob-setup/.venv/lib/python3.8/site-packages/paramiko/transport.py", line 2215, in _check_banner
out ->     raise SSHException(
out -> paramiko.ssh_exception.SSHException: Error reading SSH protocol banner
out ->
out -> Error reading ssh protocol banner. This might be a temporary error, or the server is a FTP(not SFTP) server.out ->
Sentry is attempting to send 18 pending error messages
Waiting up to 2 seconds
Press Ctrl-C to quit
```

Exactly the same output. 

Why is sentry sending so many events, supposedly splitted by each newline? 

The core issue in this paramiko example is that we're trying to connect to an FTP, instead of an SFTP. The FTP is a windows filezilla server, which when attempting connect send the response ` 500 Syntax error, command unrecognized` because paramiko only supports SFTP connections. We've patched this internally but still are dealing with the logs.

**Comment by untitaker** [comment]
Those events seem to have a `logger` tag `paramiko.transport`. Based on that I went into their sourcecode and found this: https://github.com/paramiko/paramiko/blob/ae3d0febef17a8ece5268bbf6c210a30573ce800/paramiko/transport.py#L2130

This calls `_log`, a helper function that logs each line individually using that logger, with the ERROR level, which creates a sentry event for each line.

You can ignore that logger using the mechanism described at https://docs.sentry.io/platforms/python/logging/#ignoring-a-logger, or disable event creation from error logs entirely using the information found on that page as well.

TLDR:

```python
from sentry_sdk.integrations.logging import ignore_logger
ignore_logger("paramiko.transport")
```

**Comment by Lilja** [comment]
Works great!
